src/providers/bedrock/utils.ts

```diff
@@ -241,14 +441,69 @@ export async function getAssumedRoleCredentials(
     const xmlData = await response.text();
     credentials = parseXml(xmlData);
     if (putInCacheWithValue) {
-      putInCacheWithValue(env(c), cacheKey, credentials, 60); //1 minute
+      putInCacheWithValue(env(c), cacheKey, credentials, 300); //5 minutes
     }
   } catch (error) {
     console.error({ message: `Error assuming role:, ${error}` });
   }
   return credentials;
 }
 
+export async function getAssumedRoleCredentials(
+  c: Context,
+  awsRoleArn: string,
+  awsExternalId: string,
+  awsRegion: string,
+  creds?: {
+    accessKeyId: string;
+    secretAccessKey: string;
+    sessionToken?: string;
+  }
+) {
+  let accessKeyId: string =
+    creds?.accessKeyId || env(c).AWS_ASSUME_ROLE_ACCESS_KEY_ID || '';
+  let secretAccessKey: string =
+    creds?.secretAccessKey || env(c).AWS_ASSUME_ROLE_SECRET_ACCESS_KEY || '';
+  let sessionToken = creds?.sessionToken;
+  // if not passed get from IRSA>WebAssumed>IMDS
+  if (!accessKeyId && !secretAccessKey) {
```

Comment:
Should we put a `getRuntimeKey() === "node"` check here? This will anyways not be useful for workerd runtime

Reply:
done